src/mlpack/tests/load_save_test.cpp

```diff
@@ -2532,6 +2532,132 @@ TEST_CASE("LoadCSVHeaderTest", "[LoadSaveTest]")
 // available in Armadillo.
 #if ARMA_VERSION_MAJOR >= 12
 
+TEST_CASE("LoadCSVNoHeaderTest", "[LoadSaveTest]")
+{
+  fstream f;
+  f.open("test.csv", fstream::out);
+  f << "a,b,c,d" << endl;
+  f << "1,2,3,4" << endl;
+  f << "5,6,7,8" << endl;
+
+  arma::mat dataset;
+  data::TextOptions opts;
+  opts.HasHeaders() = false;
+  opts.MissingToNan() = true;
+  REQUIRE(data::Load("test.csv", dataset, opts) == true);
+
+  REQUIRE(dataset.n_rows == 4);
+  REQUIRE(dataset.n_cols == 3);
+  REQUIRE(std::isnan(dataset.at(0, 0)) == true);
+  REQUIRE(std::isnan(dataset.at(1, 0)) == true);
+  REQUIRE(std::isnan(dataset.at(2, 0)) == true);
+  REQUIRE(std::isnan(dataset.at(3, 0)) == true);
+}
+
+#endif
+
+TEST_CASE("LoadVectorCSVFiles", "[LoadSaveTest]")
+{
+  std::vector<std::string> files = {"f0.csv", "f1.csv", "f2.csv", "f3.csv",
+      "f4.csv", "f5.csv", "f6.csv", "f7.csv", "f8.csv", "f9.csv"};
+
+  arma::mat dataset;
+  data::TextOptions opts;
+  REQUIRE(data::Load(files, dataset, opts) == true);
+
+  REQUIRE(dataset.n_rows == 5);
+  REQUIRE(dataset.n_cols == 100);
```

Comment:
```suggestion
  REQUIRE(dataset.n_cols == 100);
  REQUIRE(dataset(0, 10) == 1.0);
```
This is just a spot check to make sure we didn't load `f0.csv` ten times.